comRamona/assignment3/WordSearchTest.java

```diff
@@ -0,0 +1,60 @@
+package assignment3;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+
+public class WordSearchTest {
+    Dictionary sampleDict;
+
+    @Before
+    public void setUp() {
+        sampleDict = new Dictionary();
+        sampleDict.addWord("CAR");
+        sampleDict.addWord("CARD");
+        sampleDict.addWord("CART");
+        sampleDict.addWord("CAT");
+    }
+
+    @Test
+    public void testSimpleCase() {
+        char[][] grid = new char[][]{
+                {'A', 'A', 'R'},
+                {'T', 'C', 'D'}
+        };
+        WordSearch wordSearch = new WordSearch(sampleDict, grid);
+        Set<String> result = wordSearch.findWords();
+        Set<String> expected = new HashSet<>(Arrays.asList("CAR", "CARD", "CAT"));
+        assertEquals(expected, result);
+    }
+
+    @Test
+    public void testSameLetterTwiceInAWord() {
```

Comment:
Also adding "ARA" to the dictionary would test that the same letter cannot be used twice from the same position.